pkg/kubelet/winstats/perfcounter_nodestats.go

```diff
@@ -171,23 +189,29 @@ func (p *perfCounterNodeStatsClient) collectMetricsData(cpuCounter, memWorkingSe
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	p.nodeMetrics = nodeMetrics{
-		cpuUsageCoreNanoSeconds:   p.convertCPUValue(cpuValue),
+		cpuUsageCoreNanoSeconds:   p.convertCPUValue(cpuCores, cpuValue),
+		cpuUsageNanoCores:         p.getCPUUsageNanoCores(),
 		memoryPrivWorkingSetBytes: memWorkingSetValue,
 		memoryCommittedBytes:      memCommittedBytesValue,
 		interfaceStats:            networkAdapterStats,
 		timeStamp:                 time.Now(),
 	}
 }
 
-func (p *perfCounterNodeStatsClient) convertCPUValue(cpuValue uint64) uint64 {
-	cpuCores := runtime.NumCPU()
+func (p *perfCounterNodeStatsClient) convertCPUValue(cpuCores int, cpuValue uint64) uint64 {
 	// This converts perf counter data which is cpu percentage for all cores into nanoseconds.
 	// The formula is (cpuPercentage / 100.0) * #cores * 1e+9 (nano seconds). More info here:
 	// https://github.com/kubernetes/heapster/issues/650
 	newValue := p.nodeMetrics.cpuUsageCoreNanoSeconds + uint64((float64(cpuValue)/100.0)*float64(cpuCores)*1e9)
 	return newValue
 }
 
+func (p *perfCounterNodeStatsClient) getCPUUsageNanoCores() uint64 {
+	// UsageNanoCores is an average over the collection period
+	cpuUsageNanoCores := (p.cpuUsageCache.latestValue - p.cpuUsageCache.previousValue) / uint64(defaultCachePeriod)
```

Comment:
Since time in Go is denominated in nanoseconds (a 10 second cache period is `10*1000*1000*1000` under the hood), will this actually give a value in cores rather than nanocores? 

Assuming 10 second period:
`uncs(t1) = 1*1e9`
`uncs(t2) = 3*1e9`
`uncs(t2) - uncs(t1) / t2 - t1 = 2*1e9 / 10*1e9 = 0.2` but with integer division it's just 0.

Reply:
Thanks for pointing out this! I wasn't quite familiar with Go data types and I just fixed the calculation to 
```
cpuUsageNanoCores := (p.cpuUsageCache.latestValue - p.cpuUsageCache.previousValue) * uint64(time.Second / time.Nanosecond) / uint64(defaultCachePeriod)
```
And the windows unit test passed with the expected value `uint64(300000000)` for `cpuUsageCache{latestValue: uint64(5000000000), previousValue: uint64(2000000000),}`.
```
PASS
ok      _/C_/Windows/system32/kubernetes/pkg/kubelet/winstats      0.079s
```